### PR TITLE
Pass --debug to kompile in kbuild

### PR DESCRIPTION
**Pull request by virgil-serbanuta**
Fixes https://github.com/runtimeverification/pyk/issues/1045

### CODE REVIEW

File: src/pyk/kbuild/__main__.py [hunk 1]
```diff
@@ -41,6 +41,12 @@ def _argument_parser() -> ArgumentParser:
         type=Path,
         help='output directory',
     )
+    kompile_parser.add_argument(
+        '--debug',
+        action=BooleanOptionalAction,
+        dest='debug',
```
Comment (tothtamas28) [inline src/pyk/kbuild/__main__.py:47]: `dest` can probably be omitted.

File: src/pyk/kbuild/__main__.py [hunk 2]
```diff
@@ -41,6 +41,12 @@ def _argument_parser() -> ArgumentParser:
...
+        help='send --debug to kompile',
```
Comment (tothtamas28) [inline src/pyk/kbuild/__main__.py:48]: How is it displayed for a `BooleanOptionalAction`?

File: src/pyk/kbuild/__main__.py [hunk 3]
```diff
@@ -60,13 +66,13 @@ def _project_file(start_dir: Path) -> Path:
     return find_file_upwards(PROJECT_FILE_NAME, start_dir)
 
 
-def do_kompile(start_dir: Path, target_name: str, output_dir: Path | None, **kwargs: Any) -> None:
+def do_kompile(start_dir: Path, target_name: str, output_dir: Path | None, debug: bool, **kwargs: Any) -> None:
```
Comment (tothtamas28) [inline src/pyk/kbuild/__main__.py:69]: Is `debug` never `None`?

File: src/pyk/kbuild/kbuild.py [hunk 4]
```diff
@@ -38,7 +38,7 @@ def k_version(self) -> str:
     def definition_dir(self, project: Project, target_name: str) -> Path:
         return self.kdist_dir / self.k_version / target_name
 
-    def kompile(self, project: Project, target_name: str) -> Path:
+    def kompile(self, project: Project, target_name: str, debug: bool = False) -> Path:
```
Comment (tothtamas28) [inline src/pyk/kbuild/kbuild.py:41]:
```suggestion
    def kompile(self, project: Project, target_name: str, *, debug: bool = False) -> Path:
```

File: src/pyk/kbuild/kbuild.py [hunk 5]
```diff
@@ -88,12 +88,13 @@ def sync(self) -> None:
         for sub_project in self.project.sub_projects:
             self._sync_project(sub_project)
 
-    def kompile(self, target_name: str, output_dir: Path) -> None:
+    def kompile(self, target_name: str, output_dir: Path, debug: bool = False) -> None:
```
Comment (tothtamas28) [inline src/pyk/kbuild/kbuild.py:91]:
```suggestion
    def kompile(self, target_name: str, output_dir: Path, *, debug: bool = False) -> None:
```

File: src/pyk/kbuild/__main__.py (hunk 2 above)
Comment (virgil-serbanuta) [reply src/pyk/kbuild/__main__.py:48]: Missing argument:
```
usage: kbuild kompile [-h] [-o OUTPUT] [--debug | --no-debug] TARGET
kbuild kompile: error: the following arguments are required: TARGET
```

`--help`:
```
usage: kbuild kompile [-h] [-o OUTPUT] [--debug | --no-debug] TARGET

positional arguments:
  TARGET                target to build

options:
  -h, --help            show this help message and exit
  -o OUTPUT, --output OUTPUT
                        output directory
  --debug, --no-debug   send --debug to kompile
```

I'll change the text to "whether to send --debug to kompile"

File: src/pyk/kbuild/__main__.py (hunk 1 above)
Comment (virgil-serbanuta) [reply src/pyk/kbuild/__main__.py:47]: Done

File: src/pyk/kbuild/__main__.py (hunk 3 above)
Comment (virgil-serbanuta) [reply src/pyk/kbuild/__main__.py:70]: You're right, it can be `None`, fixed by adding a default value.

File: src/pyk/kbuild/__main__.py (hunk 1 above)
Comment (tothtamas28) [reply src/pyk/kbuild/__main__.py:47]: `dest` is still set, consider dropping it.

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**